contrib/src/main/scala/akka/stream/contrib/FocusedFlow.scala

```diff
@@ -0,0 +1,241 @@
+package akka.stream.contrib
+
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.stream._
+import akka.stream.contrib.FocusedSource.IncompleteFocusedSourceShape
+import akka.stream.scaladsl.{BidiFlow, Flow, FlowOpsMat, GraphDSL, Keep, RunnableGraph, Sink, Source}
+import akka.stream.stage.{GraphStage, GraphStageLogic, InHandler, OutHandler}
+
+import scala.annotation.unchecked.uncheckedVariance
+import scala.collection.immutable
+
+object FocusedFlow {
+  implicit class FocusedFlowOps[Tag, Src, In, Mat](from: Flow[Src, (Tag, In), Mat]) {
+    def focus = new FocusedFlow(BidiFlow.fromFlowsMat(from, Flow[In])(Keep.left))
+  }
+
+  def main(args: Array[String]): Unit = {
+    implicit val system = ActorSystem()
+    implicit val materializer = ActorMaterializer()
+
+    val flow: Flow[(Int, String), (Int, String), NotUsed] = Flow[(Int, String)]
+      .focus
+        .map(_.toUpperCase)
+        .filterNot(_.startsWith("T"))
```

Comment:
Another question, will `grouped` work, and what behaviour would you expect from it?

Use case: Lagom offers Kafka streams as ordinary streams of messages with no envelope, the user supplies a `Flow[T, Done, _]` to handle it, and the emission of each `Done` causes the messages to be committed (this itself may be batched, but that's irrelevant here).  So Lagom would do this:

```scala
kafkaConsumer
  .focus
  .via(userSuppliedFlow)
  .unfocus
  .map(commit)
```

Now, if the user wants to batch processing themselves, for efficiency reasons, then they might use `Flow.groupedWithin` followed by a `mapAsync`. In the best case scenario, we could allow them to return a single `Done` for the batch, and the unfocus would just select either the first or the last message to associate that `Done` with. Alternatively, the user might have to do a `mapConcat` to ensure that they still emit one message for each message that was grouped.